File: backend/app/Models/ContractEmployeePosition.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class ContractEmployeePosition extends Model
{
    use HasFactory;

    protected $table = 'contract_employee_position';

    protected $fillable = [
        'employee_id',
        'contract_id',
        'position_id',
        'is_main',
        'start_date',
        'end_date',
        'base_salary',
        'ratio',
    ];

    public function employee(): BelongsTo
    {
        return $this->belongsTo(Employee::class, 'employee_id');
    }

    public function contract(): BelongsTo
    {
        return $this->belongsTo(Contract::class, 'contract_id');
    }

    public function position(): BelongsTo
    {
        return $this->belongsTo(Position::class, 'position_id');
    }
}

File: hr_management_app/lib/models/contract_employee_position.dart
class ContractEmployeePosition {
  final int employeeId;
  final int contractId;
  final int positionId;
  final bool isMain;
  final String startDate;
  final String? endDate;
  final double baseSalary;
  final double ratio;
  final String createdAt;
  final String updatedAt;

  ContractEmployeePosition({
    required this.employeeId,
    required this.contractId,
    required this.positionId,
    required this.isMain,
    required this.startDate,
    this.endDate,
    required this.baseSalary,
    required this.ratio,
    required this.createdAt,
    required this.updatedAt,
  });

  factory ContractEmployeePosition.fromJson(Map<String, dynamic> json) {
    return ContractEmployeePosition(
      employeeId: json['employee_id'] ?? 0,
      contractId: json['contract_id'] ?? 0,
      positionId: json['position_id'] ?? 0,
      isMain: json['is_main'] ?? false,
      startDate: json['start_date'] ?? '',
      endDate: json['end_date'],
      baseSalary: (json['base_salary'] ?? 0.0).toDouble(),
      ratio: (json['ratio'] ?? 0.0).toDouble(),
      createdAt: json['created_at'] ?? '',
      updatedAt: json['updated_at'] ?? '',
    );
  }
}